common/src/main/java/com/wynntils/wc/model/CharacterManager.java

```diff
@@ -25,9 +27,23 @@ public class CharacterManager extends CoreManager {
     private static final Pattern CLASS_PATTERN = Pattern.compile("§e- §r§7Class: §r§f(.+)");
     private static final Pattern LEVEL_PATTERN = Pattern.compile("§e- §r§7Level: §r§f(\\d+)");
 
+    private static final int[] LEVEL_UP_XP_REQUIREMENTS = {
+        110, 190, 275, 385, 505, 645, 790, 940, 1100, 1370, 1570, 1800, 2090, 2400, 2720, 3100, 3600, 4150, 4800, 5300,
+        5900, 6750, 7750, 8900, 10200, 11650, 13300, 15200, 17150, 19600, 22100, 24900, 28000, 31500, 35500, 39900,
+        44700, 50000, 55800, 62000, 68800, 76400, 84700, 93800, 103800, 114800, 126800, 140000, 154500, 170300, 187600,
+        206500, 227000, 249500, 274000, 300500, 329500, 361000, 395000, 432200, 472300, 515800, 562800, 613700, 668600,
+        728000, 792000, 860000, 935000, 1040400, 1154400, 1282600, 1414800, 1567500, 1730400, 1837000, 1954800, 2077600,
+        2194400, 2325600, 2455000, 2645000, 2845000, 3141100, 3404710, 3782160, 4151400, 4604100, 5057300, 5533840,
+        6087120, 6685120, 7352800, 8080800, 8725600, 9578400, 10545600, 11585600, 12740000, 14418250, 16280000,
+        21196500, 23315500, 25649000, 249232940
```

Comment:
Maybe we should calculate the 106 xp requirement instead of hardcoding in case values change in the future?

Reply:
Oh, oh... You don't know what hornet's nest you're stirring! :) I've spent a lot of time with spread sheets, reverse engineering formulas, and manual testing. Overall, Wynncrafts functions can be quite erratic. 

I understand your thinking, but I think we do best to keep it hard-coded. 

I can see if I can dig out some links to how these values where extracted, though.

Reply:
I believe 1-105 is all hardcoded without much formula, 106 should just be the sum of everything 1-105

Reply:
I see. It does not seem worth it to calculate it. If values are changed, we still need to update the array. I added a comment about this observation, though.